ion-c-sys/src/writer.rs

```diff
@@ -15,6 +15,41 @@ pub enum WriterMode {
     Binary = 1,
 }
 
+/// The API for Ion C value writing.
+pub trait IonCValueWriter {
+    /// Writes a `null` value.
+    fn write_null(&mut self, tid: ION_TYPE) -> IonCResult<()>;
+
+    /// Writes a `bool` value.
+    fn write_bool(&mut self, value: bool) -> IonCResult<()>;
+
+    /// Writes an `int` value.
+    fn write_i64(&mut self, value: i64) -> IonCResult<()>;
+
+    // TODO ion-rust/#50 - support ION_INT (arbitrary large integer) writes
+
+    /// Writes a `float` value.
+    fn write_f64(&mut self, value: f64) -> IonCResult<()>;
+
+    // TODO ion-rust/#42 - support decimal writes
+    // TODO ion-rust/#43 - support timestamp writes
+
+    /// Writes a `symbol` value.
+    fn write_symbol(&mut self, value: &str) -> IonCResult<()>;
+
+    /// Writes a `string` value.
+    fn write_string(&mut self, value: &str) -> IonCResult<()>;
+
+    /// Writes a `clob` value.
+    fn write_clob(&mut self, value: &[u8]) -> IonCResult<()>;
+
+    /// Writes a `blob` value.
+    fn write_blob(&mut self, value: &[u8]) -> IonCResult<()>;
+
+    /// Starts  a container.
+    fn start_container(&mut self, tid: ION_TYPE) -> IonCResult<()>;
```

Comment:
Don't we need an `end_container` here, or did I miss something?

Reply:
I didn't add it because it is not relevant to all of the writing contexts, though I think that is potentially confusing.